### PR TITLE
Fix user listing error (2nd try)

### PR DESCRIPTION
**Pull request by hypery2k**
FIXED #JENKINS-18884

New clean PR for #3629

See [JENKINS-18884](https://issues.jenkins-ci.org/browse/JENKINS-18884).

And People page of All view also displays all users registered in Jenkins.

They do no permission check.

This patch does followings:

Adds permission check to jobs to extract users (now requires Read permission).
People page of All view no longer display all users registered in Jenkins. It works same as those of other views:
There is no reason for People page of All page to work different from those of other views.
Displaying all users in Jenkins should be handled by security realms (like HudsonPrivateSecurityRealm.ManageUserLinks). 
This is another approach of #1094 .

Readdition of #1102
### Proposed changelog entries

* View of users requiring overall read permission

### Submitter checklist

- [x] JIRA issue is well described
- [x] Changelog entry appropriate for the audience affected by the change (users or developer, depending on the change). [Examples](https://github.com/jenkins-infra/jenkins.io/blob/master/content/_data/changelogs/weekly.yml)
      * Use the `Internal: ` prefix if the change has no user-visible impact (API, test frameworks, etc.)
- [x] Appropriate autotests or explanation to why this change has no tests
- [ ] For dependency updates: links to external changelogs and, if possible, full diffs

### Desired reviewers

@oleg-nenashev

### CODE REVIEW

**Comment by jeffret-b** [comment]
The tests have some problems.

1. The path to the test resources uses "AsyncPeopleTest" instead of "AsynchPeopleTest" so the resources cannot be found.
2. The `@LocalData` also needs to supply a "users/users.xml" mapping file that references the two users and their directories. See "hudson/security/ExtendedReadPermissionTest/users/users.xml" for one example.

**Comment by hypery2k** [comment]
thanks for the hint.

**Comment by jeffret-b** [comment]
> any chance to get locally the unit test running? Always getting errors from other Unit Tests

I'm not sure what you're asking. I tried building and running the tests on my machine and I'm getting the same failure as on the CI job.

File: test/src/test/resources/hudson/model/AsynchPeopleTest/users/admin/config.xml [hunk 1]
```diff
@@ -0,0 +1,10 @@
+<?xml version='1.0' encoding='UTF-8'?>
+<user>
+    <fullName>admin</fullName>
+    <properties>
+        <hudson.security.HudsonPrivateSecurityRealm_-Details>
+            <!--password: admin-->
+            <passwordHash>#jbcrypt:$2a$10$eBXjXOF.vPfWeqmYlrsSdeupTlcjh5qAOCeKqoH2JWQGcnE3vfR9K</passwordHash>
+        </hudson.security.HudsonPrivateSecurityRealm_-Details>
+    </properties>
+</user>
```
Comment (jsoref) [inline test/src/test/resources/hudson/model/AsynchPeopleTest/users/admin/config.xml]: is there a reason some of these files don't have EOL at EOF?

**Comment by alecharp** [comment]
@hypery2k would you have time to revisit this PR?

**Comment by hypery2k** [comment]
not really currently, sorry

**Comment by varyvol** [comment]
@hypery2k there is no activity since August, so marking this as proposed for close. It will be closed in the next pass. Please reach out in case you are still interested in working on this.

**Comment by rsandell** [comment]
Closing because no activity. You are welcome to reopen when work is resumed.

**Pull request closed without merging**